### PR TITLE
[Security] Removed security-acl from the core

### PR DESCRIPTION
**Pull request by linaori**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | part of #14718 |
| License | MIT |
| Doc PR | ~ |

The `Security\Acl` is removed from the core and is loaded from its own repository. All tests were passing and this is fully backwards compatible. I have removed all but the Test files in the first step and added the dependency to verify the Test were still working with the package dependency. The second step was to remove the remaining test files and tests are still running for both the Bundle and the Framework. Once the Read-Only repository is a full standalone repository, this PR can be merged. 
- [x] Remove component from the core
- [x] Remove read-only from https://github.com/symfony/security-acl

Once this PR is merged, I can start working on splitting the SecurityBundle and extracting the ACL part to the AclBundle.

/cc @fabpot

### CODE REVIEW

**Comment by stof** [comment]
``symfony/security`` should not replace ``symfony/security-acl`` anymore either

File: src/Symfony/Bundle/SecurityBundle/composer.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "require": {
         "php": ">=5.3.9",
         "symfony/security": "~2.8|~3.0.0",
-        "symfony/http-kernel": "~2.2|~3.0.0"
+        "symfony/http-kernel": "~2.2|~3.0.0",
+        "symfony/security-acl": "~2.8|~3.0.0"
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/composer.json]: should it really be a required dependency ?

Comment (iltar) [inline src/Symfony/Bundle/SecurityBundle/composer.json]: To preserve backwards compatibility as it's always available atm. Once the Acl Bundle is created, I can move this dependency there and put the bundle as dependency in the SecurityBundle. In 3.0 the BC layer (regarding config) + dependency can be removed.

**Comment by iltar** [comment]
@stof ah yes, I will update those accordingly

**Comment by stof** [comment]
btw, the not-removed replacement is the reason why tests are failing

**Comment by iltar** [comment]
Lets see what happens after a push. For some reason my 2.8 keeps complaining about objects not being equal in form tests;
```
73) Symfony\Component\Validator\Tests\Constraints\RangeValidatorTest::testInvalidDatesCombinedMin with data set #3 (DateTimeImmutable Object (...), 'Mar 9, 2014, 12:00 AM')
Failed asserting that two objects are equal.
--- Expected
+++ Actual
@@ @@
 Symfony\Component\Validator\ConstraintViolation Object (
     'message' => null
     'messageTemplate' => 'myMinMessage'
     'parameters' => Array (
-        '{{ value }}' => 'Mar 9, 2014, 12:00 AM'
-        '{{ limit }}' => 'Mar 10, 2014, 12:00 AM'
+        '{{ value }}' => 'Mar 9, 2014 12:00 AM'
+        '{{ limit }}' => 'Mar 10, 2014 12:00 AM'
     )
     'plural' => null
     'root' => 'root'
     'propertyPath' => 'property.path'
     'invalidValue' => 'InvalidValue'
     'constraint' => Symfony\Component\Validator\Constraints\NotNull Object (...)
     'code' => 3
     'cause' => null
 )

```

**Comment by fabpot** [comment]
Thank you @iltar.

**Comment by fabpot** [comment]
As of today, the ACL subtree split is not managed by the splitter anymore as of branch 2.8. For older branches, it is still active, even if there are close to no activity on the component. I've also activated the ticketing system on the repo.

**Comment by fabpot** [comment]
@iltar The next step would be to create the AclBundle and remove everything related to Acl in core (you had some good ideas in the linked issue).

**Comment by iltar** [comment]
Thanks! I will get on it.

**Pull request closed without merging**